dotCMS/src/main/java/com/dotcms/rendering/velocity/services/PageLoader.java

```diff
@@ -0,0 +1,212 @@
+import com.dotmarketing.portlets.contentlet.business.ContentletAPI;
+import com.dotmarketing.portlets.contentlet.model.Contentlet;
+import com.dotmarketing.portlets.htmlpageasset.model.HTMLPageAsset;
+import com.dotmarketing.portlets.htmlpageasset.model.IHTMLPage;
+import com.dotmarketing.portlets.templates.model.Template;
+import com.dotmarketing.tag.model.Tag;
+import com.dotmarketing.util.Config;
+import com.dotmarketing.util.InodeUtils;
+import com.dotmarketing.util.Logger;
+import com.dotmarketing.util.PageMode;
+import com.dotmarketing.util.TagUtil;
+
+import java.io.File;
+import java.io.InputStream;
+import java.util.ArrayList;
+import java.util.List;
+
+import org.apache.velocity.runtime.resource.ResourceManager;
+
+import com.liferay.portal.model.User;
+
+/**
+ * @author will
+ *
+ *         To change this generated comment edit the template variable "typecomment":
+ *         Window>Preferences>Java>Templates. To enable and disable the creation of type comments go
+ *         to Window>Preferences>Java>Code Generation.
+ */
+public class PageLoader implements DotLoader {
+
+
+
+    @Override
+    public void invalidate(Object obj) {
+        IHTMLPage htmlPage = (IHTMLPage) obj;
+        for (PageMode mode : PageMode.values()) {
+            invalidate(htmlPage, mode);
+        }
+
+
+    }
+
+    @Override
+    public void invalidate(Object obj, PageMode mode) {
+        IHTMLPage htmlPage = (IHTMLPage) obj;
+
+
+        String folderPath = mode.name() + File.separator;
+        String velocityRootPath = VelocityUtil.getVelocityRootPath();
+        String languageStr = htmlPage.isContent() ? "_" + ((Contentlet) htmlPage).getLanguageId() : "";
+        String filePath = folderPath + htmlPage.getIdentifier() + languageStr + "." + VelocityType.HTMLPAGE.fileExtension;
+        velocityRootPath += java.io.File.separator;
+        java.io.File f = new java.io.File(velocityRootPath + filePath);
+        f.delete();
+        DotResourceCache vc = CacheLocator.getVeloctyResourceCache();
+        vc.remove(ResourceManager.RESOURCE_TEMPLATE + filePath);
+
+    }
+
+
+
+    public InputStream buildStream(IHTMLPage htmlPage, PageMode mode, final String filePath)
+            throws DotStateException, DotDataException {
+        Identifier identifier = APILocator.getIdentifierAPI()
+            .find(htmlPage);
+        try {
+            return buildStream(htmlPage, identifier, mode, filePath);
+        } catch (Exception e) {
+            Logger.error(this.getClass(), e.getMessage(), e);
+            throw new DotRuntimeException(e.getMessage());
+        }
+    }
+
+
+    public InputStream buildStream(IHTMLPage htmlPage, Identifier identifier, PageMode mode, final String filePath)
+            throws DotDataException, DotSecurityException {
+        String folderPath = mode.name() + File.separator;
+        InputStream result;
+        StringBuilder sb = new StringBuilder();
+
+        ContentletAPI conAPI = APILocator.getContentletAPI();
```

Comment:
![MAJOR](https://sonarsource.github.io/sonar-github/severity-major.png 'Severity: MAJOR') Remove this useless assignment to local variable "conAPI". [![rule](https://sonarsource.github.io/sonar-github/rule.png)](https://sonarcloud.io/coding_rules#rule_key=squid%3AS1854)
![MINOR](https://sonarsource.github.io/sonar-github/severity-minor.png 'Severity: MINOR') Remove this unused "conAPI" local variable. [![rule](https://sonarsource.github.io/sonar-github/rule.png)](https://sonarcloud.io/coding_rules#rule_key=squid%3AS1481)